File: Code/0203.Remove Linked List Elements/Remove Linked List Elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        if not head:
            return head
        # 当要删除的元素为第一个元素时
        while head and head.val == val:
            head = head.next
        else:
            cur = head
            pre = None              # 用于储存被删除节点的上一节点
            while cur is not None:  # 遍历到尾节点
                if cur.val != val:
                    pre = cur
                else:
                    pre.next = cur.next
                cur = cur.next
        return head

File: Code/0203.Remove Linked List Elements/Remove Linked List Elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        // 当要删除的节点为第一个元素时
        while(head != null && head.val == val){
            head = head.next;
        }
        if(head == null){
            return null;
        }
        ListNode cur = head;
        ListNode pre = null;

        while(cur != null){
            if(cur.val != val){
                pre = cur;
            }else{
                pre.next = cur.next;
            }
            cur = cur.next;
        }

        return head;

    }
}